src/pages/workspace/WorkspaceSidebar.js

```diff
@@ -65,14 +73,15 @@ const WorkspaceSidebar = ({translate, isSmallScreenWidth, policy}) => {
         },
     ];
 
-    useEffect(() => {
-        if (_.isEmpty(policy)) {
-            Growl.error(translate('workspace.error.growlMessageInvalidPolicy'), CONST.GROWL.DURATION_LONG);
-            Navigation.dismissModal();
-            create();
-            return null;
-        }
-    }, [policy]);
+    // After all the policies have loaded, we can know if the given policyID points to a nonexistant workspace
+    // When free plan is out of beta and Permissions.canUseFreePlan() gets removed,
+    // all code involving 'allPolicies' can be removed since policy loading will no longer be delayed on login.
```

Comment:
NAB: I feel like it would be nice to use dome keyword like `TODO:` in places where we _**know for certain**_ we will make a change in future to maybe keep track of it and find it more easily in future. WDYT? 

Of course, if we would decide to use some keyword, then it will be properly useful only if we use it in all the places like this. Let me know what you think about this?

Reply:
I like TODO because if we miss it, someone will definitely notice it's off.

Reply:
Yes, let's add a `// @TODO` (this is our standard format) and then also create a GH issue to do the removal and make sure it happens.

Reply:
I created a GH issue and also made a note to reexamine if the two components LoginWith can be consolidated. I think they can be and it'll be good to reexamine when this is out of beta.

Reply:
I'm not sure if everyone still agrees with this but we've historically discouraged the use of TODO comments and just use GH issues. TODO comments tend to live forever in the code, and pollute the code as they become outdated, or because there's usually no timeline for when they need to be done, etc. 

Is there anything that the GH issue doesn't solve, that we need the TODO comment for?

Reply:
![image](https://user-images.githubusercontent.com/165133/129953611-5814ee06-1eee-4e16-bbec-dff8012d9d3d.png)
![image](https://user-images.githubusercontent.com/165133/129953663-2ed43a1e-992f-41c7-8998-c2d30226f10c.png)
![image](https://user-images.githubusercontent.com/165133/129953707-b4e66df4-6914-4915-a1d6-32d3bf96e8ca.png)
![image](https://user-images.githubusercontent.com/165133/129953815-405961a0-b7b3-4100-8ca7-b91ecc9e7929.png)

Reply:
Wow, interesting, did not realize this at all. If there is such an experience with using it, it might good decision to just omit it this time and we will make the change needed once the time comes.

Reply:
Ah I wish github would bump down these comments so they're easier to see. Since we have the issue created I don't feel strongly about keepign it as as `TODO`. I'll remove it.